frontend/src/components/translation/TranslationTable.tsx

```diff
@@ -38,8 +67,45 @@ const TranslationTable = ({
     setCommentLine(displayLineNumber);
     setCommentStoryTranslationContentId(storyTranslationContentId);
   };
+  const [updateStatus] = useMutation<{
+    response: UpdateStoryTranslationContentStatusResponse;
+  }>(UPDATE_STORY_TRANSLATION_CONTENT_STATUS);
+
+  const handleStatusChange = async (storyLine: StoryLine, newState: string) => {
+    const result = updateStatus({
+      variables: {
+        storyTranslationId: storyLine.storyTranslationContentId!!,
+        status: newState,
+      },
+    });
+    if (result) {
+      const prevState = translatedStoryLines[storyLine.lineIndex].status;
+
+      const updatedStatusArray = [...translatedStoryLines!];
+      updatedStatusArray[storyLine.lineIndex].status = convertStatusTitleCase(
+        newState,
+      );
+
+      await setTranslatedStoryLines!!(updatedStatusArray);
```

Comment:
without the `!!` the code won't compile because setTranslatedStoryLines is an optional parameter as the TranslationPage does not pass it in when it calls TranslationTable. 

Tho, if I moved the Menu component to its own file, that might resolve the problem.